compose/snippets/src/main/java/com/example/compose/snippets/animations/sharedelement/BasicSharedElementSnippets.kt

```diff
@@ -441,3 +443,57 @@ private object UniqueKeySnippet {
     }
     // [END android_compose_shared_elements_unique_key]
 }
+
+// [START android_compose_shared_element_scope]
+val LocalNavAnimatedVisibilityScope = compositionLocalOf<AnimatedVisibilityScope?> { null }
+val LocalSharedTransitionScope = compositionLocalOf<SharedTransitionScope?> { null }
+
+@Composable
+private fun SharedElementScope_CompositionLocal() {
+    // An example of how to use composition locals to pass around the shared transition scope, far down your UI tree.
+    // [START_EXCLUDE]
+    val state = remember {
+        mutableStateOf(false)
+    }
+    // [END_EXCLUDE]
+    SharedTransitionLayout {
+        CompositionLocalProvider(
+            LocalSharedTransitionScope provides this
+        ) {
+            // This could also be your top-level NavHost as this provides an AnimatedContentScope
+            AnimatedContent(state, label = "Top level AnimatedContent") { targetState ->
+                CompositionLocalProvider(LocalNavAnimatedVisibilityScope provides this) {
+                    // Now we can access the scopes in any nested composables as follows:
+                    val sharedTransitionScope = LocalSharedTransitionScope.current
+                        ?: throw IllegalStateException("No SharedElementScope found")
+                    val animatedVisibilityScope = LocalNavAnimatedVisibilityScope.current
+                        ?: throw IllegalStateException("No SharedElementScope found")
```

Comment:
Should the error mention AnimatedVisibilityScope?

Reply:
yes good catch!